nemo/collections/nlp/models/language_modeling/megatron_gpt_sft_model.py

```diff
@@ -89,10 +89,8 @@ def __init__(self, cfg: DictConfig, trainer: Trainer):
             if hasattr(self.cfg.data.test_ds, "metric"):
                 self.test_metric_label_key = self.cfg.data.test_ds.metric.get('label_key', 'labels')
 
-        if self.cfg.get('megatron_amp_O2', False):
-            base_module = self.model.module
-        else:
-            base_module = self.model
+        if self.use_peft and self.cfg.get('virtual_pipeline_model_parallel_size', None):
+            raise ValueError('Virtual pipeline model parallel is not supported when using PEFT')
 
         self._reset_activation_checkpointing_args()
         self._reset_sequence_parallelism_args()
```

Comment:
I guess we don't need to do this in this PR but we need to remove this from constructor or SP will not work with SFT.

Reply:
Handled in https://github.com/NVIDIA/NeMo/pull/7963